pages/advanced.tsx

```diff
@@ -39,37 +38,13 @@ function Dropdown(props: DropdownProp) {
 
 function FilteredPokemonList(props: FilteredPokemonProp) {
   if (props.typeOfSearch == "1") {
-    return (
-      <SearchFilter
-        choice={props.searchInputValue}
-        offset={props.offset}
-        setPages={props.setPages}
-      />
-    );
+    return <SearchFilter choice={props.searchInputValue} />;
   } else if (props.typeOfSearch == "2") {
-    return (
-      <TypeFilter
-        choice={props.searchInputValue}
-        offset={props.offset}
-        setPages={props.setPages}
-      />
-    );
+    return <TypeFilter choice={props.searchInputValue} />;
   } else if (props.typeOfSearch == "3") {
-    return (
-      <AbilityFilter
-        choice={props.searchInputValue}
-        offset={props.offset}
-        setPages={props.setPages}
-      />
-    );
+    return <AbilityFilter choice={props.searchInputValue} />;
   } else if (props.typeOfSearch == "4") {
-    return (
-      <EggFilter
-        choice={props.searchInputValue}
-        offset={props.offset}
-        setPages={props.setPages}
-      />
-    );
+    return <EggFilter choice={props.searchInputValue} />;
   } else {
     return <></>;
   }
```

Comment:
Else if blocks can benefit from using what's called the ["implicit return pattern"](https://medium.com/@rachelchervin/if-else-vs-if-return-using-implicit-returns-in-javascript-and-react-bbac63656e1b). When you return, you are ending the execution of that function scope. In other words, we don't need the else keyword. If a condition is met, simply return.